assets/src/design-system/components/slider/index.js

```diff
@@ -190,11 +223,11 @@ function Slider({
   const percentageVal = ((value - min) / (max - min)) * 100;
 
   useEffect(() => {
-    if (ref.current && !widthTracker.current) {
+    if (ref.current && !widthTracker) {
       const { width } = ref.current.getBoundingClientRect();
-      widthTracker.current = width;
+      setWidthTracker(width);
     }
-  }, [ref]);
+  }, [ref, widthTracker]);
```

Comment:
I don't think either dependency is required here - and the entire `if` is also useless, I think. This `useEffect` will only effectively run once with or without dependencies and `if` – and I think that's fine. Perhaps it should be a `useLayoutEffect`.

We might however get issues if we have a slider that updates its size and/or position while rendered, but that seems like a rare edge case.

Reply:
Hm, testing shows that this `useEffect` seems to run multiple times, also, it would run at least twice since `setWidthTracker` changes a state, no?
What the `if` is basically doing is checking that if the `ref` is set but width is not set yet, set the width. Not sure what's the harm in having this check there, WDYT?

On dependencies -- adding everything used inside the hook as a dependency is even a rule in the linters AFAIK.

Reply:
My though was to make this an empty list of dependencies. The use of `widthTracker` is only necessary because it is used (if that makes sense). Remove the `if` completely, use an empty list of dependencies and the effect runs only once and performs as expected. Also, `ref`s never need to be declared as dependencies, as they're stable (and eslint knows that).